### DISCUSSION TITLE
Sidebar integration

### DISCUSSION THREAD

**Discussion by igorschoester** [woocommerce/woocommerce#41457] [WooCommerce New Product Editor] [answered]
I would like to render a button in the navigation, here:
<img width="1061" alt="woo-block-editor" src="https://github.com/woocommerce/woocommerce/assets/35524806/ac665942-f36e-4561-9932-46830174948f">

Technically, there is a Slot available with the name `woocommerce_header_item/product`. However, I can't seem to render anything there. Is the `window.wp.components.Slot` instance used here?

Is this meant as open to the public API? If not, will there be a spot there in the future? Similar to how we have plugins in the block editor. If yes, is there documentation on this?

**Comment by mattsherman** [comment, marked as answer]
Hello @igorschoester!

I believe the reason you may have had difficulty with this is because there was recently an issue fixed that addressed filling the slots in the product editor from outside of the product editor. There was also recently another fix to provide context to product editor plugins.


- https://github.com/woocommerce/woocommerce/pull/41102
- https://github.com/woocommerce/woocommerce/pull/41106

These latest changes are in `trunk` and will be available in WooCommerce 8.4.

We do not yet have documentation on using these extension points (more documentation and UX guidelines will be coming soon!). However, here is a quick example of adding a button to the header, and adding a menu item to the more (...) menu.

```php
<?php

add_action(
	'admin_enqueue_scripts',
	function() {
		wp_enqueue_script(
			'your-prefix-header',
			plugin_dir_url( __FILE__ ) . 'your-prefix-header.js',
			[],
			filemtime( dirname( __FILE__ ) . '/your-prefix-header.js' ),
			true,
		);
	}
);
```

```js
wp.plugins.registerPlugin( 'your-prefix-header', {
    scope: 'woocommerce-product-block-editor',
    render: function() {
        const postType = wp.element.useContext( wc.productEditor.PostTypeContext );
        const [ id ] = wp.coreData.useEntityProp( 'postType', postType, 'id' );

        return wp.element.createElement(
            wp.element.Fragment,
            {},
            wp.element.createElement(
                wc.adminLayout.WooHeaderItem,
                {
                    name: 'product',
                    order: 1000,
                },
                wp.element.createElement(
                    wp.components.Button,
                    {
                        onClick: () => {
                            alert( `Hello from button! Post type: ${ postType }, Post id: ${ id }` );
                        }
                    },
                    'My button'
                )
            ),
            wp.element.createElement(
                wc.productEditor.__experimentalWooProductMoreMenuItem,
                {
                    order: 1000,
                },
                wp.element.createElement(
                    wp.element.Fragment,
                    {},
                    wp.element.createElement(
                        wp.components.MenuItem,
                        {
                            onClick: () => {
                                alert( `Hello from menu! Post type: ${ postType }, Post id: ${ id }` );
                            }
                        },
                        'My menu item'
                    )
                )
            ),
        );
    }
} );
```

Please let us know if you have any further questions or comments on extending the product editor!

**Reply by igorschoester** [reply]
Thank you @mattsherman! 🙏  That is a big help 😄

I think I was missing the proper `scope` there mostly.
However, I like the other tips & tricks included. For example, the WooHeaderItem instead of my Slot with hardcoded name.
This gives me some more angles to explore your repo 🙌 

Looking forward to the documentation!
